include/global_object_fields.h

```diff
@@ -0,0 +1,429 @@
+#define /*0x194*/ oCollisionDistance                            OBJECT_FIELD_F32(0x43)
+#define /*0x198*/ oNumLootCoins                                 OBJECT_FIELD_S32(0x44)
+#define /*0x19C*/ oDrawingDistance                              OBJECT_FIELD_F32(0x45)
+#define /*0x1A0*/ oRoom                                         OBJECT_FIELD_S32(0x46)
+// 0x1A4 is unused, possibly related to 0x1A8 in removed macro purposes.
+#define /*0x1A8*/ oUnusedCoinParams                             OBJECT_FIELD_U32(0x48)
+// 0x1AC-0x1B2 (0x48-0x4A) are object specific and defined below the common fields.
+#define /*0x1B4*/ oWallAngle                  OBJECT_FIELD_S32(0x4B)
+#define /*0x1B8*/ oFloorType                  OBJECT_FIELD_S16(0x4C, 0)
+#define /*0x1BA*/ oFloorRoom                  OBJECT_FIELD_S16(0x4C, 1)
+#define /*0x1BC*/ oAngleToHome                OBJECT_FIELD_S32(0x4D)
+#define /*0x1C0*/ oFloor                      OBJECT_FIELD_SURFACE(0x4E)
+#define /*0x1C4*/ oDeathSound                 OBJECT_FIELD_S32(0x4F)
+#ifdef PUPPYLIGHTS
+#define /*0x1C4*/ oLightID                                      OBJECT_FIELD_S32(0x50)
+#endif
+
+/* Pathed (see obj_follow_path) */
+#define /*0x0FC*/ oPathedStartWaypoint     OBJECT_FIELD_WAYPOINT(0x1D)
+#define /*0x100*/ oPathedPrevWaypoint      OBJECT_FIELD_WAYPOINT(0x1E)
+#define /*0x104*/ oPathedPrevWaypointFlags OBJECT_FIELD_S32(0x1F)
+#define /*0x108*/ oPathedTargetPitch       OBJECT_FIELD_S32(0x20)
+#define /*0x10C*/ oPathedTargetYaw         OBJECT_FIELD_S32(0x21)
+
+/* Special Object Macro */
+#define /*0x108*/ oMacroUnk108 OBJECT_FIELD_F32(0x20)
+#define /*0x10C*/ oMacroUnk10C OBJECT_FIELD_F32(0x21)
+#define /*0x110*/ oMacroUnk110 OBJECT_FIELD_F32(0x22)
+
+/**
+ * These are object-specific but used globally in code
+ **/
+
+/* Mario */
+#define /*0x0F4*/ oMarioParticleFlags    OBJECT_FIELD_S32(0x1B)
+#define /*0x108*/ oMarioReadingSignDYaw  OBJECT_FIELD_S32(0x20)
+#define /*0x10C*/ oMarioCannonObjectYaw  OBJECT_FIELD_S32(0x21)
+#define /*0x10C*/ oMarioTornadoYawVel    OBJECT_FIELD_S32(0x21)
+#define /*0x10C*/ oMarioReadingSignDPosX OBJECT_FIELD_F32(0x21)
+#define /*0x110*/ oMarioReadingSignDPosZ OBJECT_FIELD_F32(0x22)
+#define /*0x110*/ oMarioPolePos          OBJECT_FIELD_F32(0x22)
+#define /*0x110*/ oMarioCannonInputYaw   OBJECT_FIELD_S32(0x22)
+#define /*0x110*/ oMarioTornadoPosY      OBJECT_FIELD_F32(0x22)
+#define /*0x110*/ oMarioWhirlpoolPosY    OBJECT_FIELD_F32(0x22)
+#define /*0x110*/ oMarioJumboStarCutscenePosZ OBJECT_FIELD_F32(0x22)
+#define /*0x110*/ oMarioBurnTimer        OBJECT_FIELD_S32(0x22)
+#define /*0x110*/ oMarioLongJumpIsSlow   OBJECT_FIELD_S32(0x22)
+#define /*0x110*/ oMarioSteepJumpYaw     OBJECT_FIELD_S32(0x22)
+#define /*0x110*/ oMarioWalkingPitch     OBJECT_FIELD_S32(0x22)
+
+/* Bowser */
+#define /*0x088*/ oBowserCamAct          OBJECT_FIELD_S32(0x00)
+#define /*0x0F4*/ oBowserStatus          OBJECT_FIELD_S32(0x1B)
+
+/* Hoot */
```

Comment:
still a bunch of specific ones here... if this is because they're referenced by other behaviors seems like a better solution is necessary